lib/puppet/functions/simplib/ip/family_hash.rb

```diff
@@ -0,0 +1,82 @@
+# Process an array of IP addresses and return them split by IP family and
+# include metadata and/or processed versions.
+#
+Puppet::Functions.create_function(:'simplib::ip::family_hash') do
+  # @param ip_addresses
+  #   The addresses to convert
+  #
+  # @return [Hash]
+  #   Converted Hash with the following format (YAML representation):
+  #
+  #   ```
+  #   # IPv4 Addresses
+  #   ipv4:
```

Comment:
Why are these grouped by family?  Is there a case in which the user would prefer to have the family set in the individual entries?

Reply:
Because this is a `family` hash. If someone wants something else, this isn't the data structure for them.

Reply:
Current usage is here: https://github.com/simp/pupmod-simp-iptables/pull/72/files#diff-57780fa042c02a1b6068fb93b67fd45dR163-R176